main.cpp

```
@@ -0,0 +1,59 @@
+#include "RegistryKey.h"
+#include <iostream>
+#include <memory>
+#include <synchapi.h>
+
+const LPCTSTR MESSAGE = "MANAGEMENT PROGRAM IS UP";
+const LPCTSTR TITLE = "Managment";
+
+const LPCSTR MUTEX_NAME = "Global\\Engineer-Mutex";
+
+// Key used to make us run on startup
+const LPCSTR TARGET_KEY = "Software\\Microsoft\\Windows\\CurrentVersion\\Run";
+
+// The field under which we will be named
+const LPCWSTR KEY_ENTRY_NAME = L"Engineer";
+
+typedef std::unique_ptr<void, decltype(&ReleaseMutex)> Mutex;
+
+/*
+ * Uses a mutex to make sure that the program is executed once.
+ * Returns a unique_ptr "Mutex" that manages the handle context and memory.
+ *
+ * name   [IN]  The name of the mutex to obtain
+ * return [OUT] A pointer to a unique_ptr for the mutex if obtained, and NULL valued unique_ptr if failed.
+ */
+Mutex executeOnce(LPCSTR mutexName) {
+    Mutex singleProgramMutex(CreateMutex(NULL, FALSE, mutexName), &ReleaseMutex);
+    if (singleProgramMutex == NULL) {
+        std::cout << "Got error in Mutex creation " << GetLastError() << std::endl;
+        return Mutex(NULL, NULL);
```

Consider throwing exceptions instead of returning invalid objects